subsys/bluetooth/services/ddfs.c

```diff
@@ -0,0 +1,416 @@
+				   (is_rtt ? sizeof(measurement->dist_estimates.rtt) : 0) +
+				   (is_mcpd ? sizeof(measurement->dist_estimates.mcpd) : 0));
+
+	dist_meas_notify = net_buf_simple_add(&buf, sizeof(*dist_meas_notify));
+	dist_meas_notify->flags = 0;
+	dist_meas_notify->quality = measurement->quality;
+	dist_meas_notify->bt_addr[6] = measurement->bt_addr.type;
+	for (uint8_t i = 0; i < 6; i++) {
+		dist_meas_notify->bt_addr[i] = measurement->bt_addr.a.val[i];
+	}
+
+	if (is_rtt) {
+		dist_meas_notify->flags |= DDF_DM_RTT_PRESENT;
+		net_buf_simple_add_le16(&buf, measurement->dist_estimates.rtt.rtt);
+	} else if (is_mcpd) {
+		dist_meas_notify->flags |= DDF_DM_MCPD_PRESENT;
+		net_buf_simple_add_le16(&buf, measurement->dist_estimates.mcpd.ifft);
+		net_buf_simple_add_le16(&buf, measurement->dist_estimates.mcpd.phase_slope);
+		net_buf_simple_add_le16(&buf, measurement->dist_estimates.mcpd.rssi_openspace);
+		net_buf_simple_add_le16(&buf, measurement->dist_estimates.mcpd.best);
+	}
+
+	if (conn) {
+		return bt_gatt_notify(conn, &ddf_svc.attrs[DDF_SVC_DISTANCE_MEAS_ATTR_IDX],
+				      dist_meas_notify, buf.len);
+	} else {
+		return bt_gatt_notify(NULL, &ddf_svc.attrs[DDF_SVC_DISTANCE_MEAS_ATTR_IDX],
+				      dist_meas_notify, buf.len);
+	}
+}
+
+int bt_ddfs_azimuth_measurement_notify(struct bt_conn *conn,
+				       const struct bt_ddfs_azimuth_measurement *measurement)
+{
+	if (!ddfs_inst.am_notification_enabled) {
+		return -EACCES;
+	}
+
+	if (!measurement) {
+		return -EINVAL;
+	}
+
+	struct ddfs_meas_notify *azimuth_meas_notify;
+
+	NET_BUF_SIMPLE_DEFINE(buf, sizeof(*azimuth_meas_notify) + sizeof(measurement->value));
+
+	azimuth_meas_notify = net_buf_simple_add(&buf, sizeof(*azimuth_meas_notify));
+	azimuth_meas_notify->flags = 0;
+	azimuth_meas_notify->quality = measurement->quality;
+	azimuth_meas_notify->bt_addr[6] = measurement->bt_addr.type;
+	for (uint8_t i = 0; i < 6; i++) {
+		azimuth_meas_notify->bt_addr[i] = measurement->bt_addr.a.val[i];
+	}
+
+	net_buf_simple_add_le16(&buf, measurement->value);
+
+	if (conn) {
+		return bt_gatt_notify(conn, &ddf_svc.attrs[DDF_SVC_AZIMUTH_MEAS_ATTR_IDX],
+				      azimuth_meas_notify, buf.len);
+	} else {
+		return bt_gatt_notify(NULL, &ddf_svc.attrs[DDF_SVC_AZIMUTH_MEAS_ATTR_IDX],
+				      azimuth_meas_notify, buf.len);
+	}
+
+	return 0;
+}
+
+int bt_ddfs_elevation_measurement_notify(struct bt_conn *conn,
+					const struct bt_ddfs_elevation_measurement *measurement)
+{
+	if (!ddfs_inst.em_notification_enebled) {
+		return -EACCES;
+	}
+
+	if (!measurement) {
+		return -EINVAL;
+	}
+
+	struct ddfs_meas_notify *elevation_meas_notify;
+
+	NET_BUF_SIMPLE_DEFINE(buf, sizeof(*elevation_meas_notify) + sizeof(measurement->value));
+
+	elevation_meas_notify = net_buf_simple_add(&buf, sizeof(*elevation_meas_notify));
+	elevation_meas_notify->flags = 0;
+	elevation_meas_notify->quality = measurement->quality;
+	elevation_meas_notify->bt_addr[6] = measurement->bt_addr.type;
+	for (uint8_t i = 0; i < 6; i++) {
```

Comment:
```suggestion
	for (size_t i = 0; i < 6; i++) {
```
Also you can add blank line before for